contract/migrations/12.0.2.0.0/pre-migration.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import SUPERUSER_ID, api
+
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
```

Comment:
Please use openupgradelib and decorate it with `@openupgrade.migrate()`. This way, you get a `env` proper populated.

You should also use `openupgrade.logged_query` method for registering the query and the result in the log, which is valuable.